File: web/src/services/chatRecords.ts
import axios from "axios";

type ApiResponse<T> = {
  code: number;
  message: string;
  data: T;
};

export type ChatRecordProvider = "chatlog" | "weflow";

export type ChatRecordSettings = {
  id: number;
  provider: ChatRecordProvider;
  chatlog_base_url: string;
  chatlog_timeout_sec: number;
  chatlog_decrypt_before_fetch: boolean;
  chatlog_decrypt_cache_enabled: boolean;
  chatlog_decrypt_timeout_sec: number;
  chatlog_decrypt_cache_buffer_sec: number;
  chatlog_work_dir: string;
  weflow_base_url: string;
  has_weflow_token: boolean;
  weflow_page_limit: number;
  weflow_page_timeout_sec: number;
  weflow_empty_page_retry: number;
  weflow_include_media: boolean;
};

export type ChatRecordSettingsPayload = Omit<ChatRecordSettings, "id" | "has_weflow_token" | "weflow_include_media"> & {
  weflow_token?: string;
};

export type ChatRecordStatus = {
  provider: ChatRecordProvider;
  status: string;
  message: string;
};

export type Chatroom = {
  name: string;
  display_name: string;
  owner?: string | null;
  user_count: number;
};

export async function fetchChatRecordSettings(): Promise<ChatRecordSettings> {
  const response = await axios.get<ApiResponse<ChatRecordSettings>>("/api/chat-records/settings");
  return response.data.data;
}

export async function updateChatRecordSettings(payload: ChatRecordSettingsPayload): Promise<ChatRecordSettings> {
  const response = await axios.put<ApiResponse<ChatRecordSettings>>("/api/chat-records/settings", payload);
  return response.data.data;
}

export async function testChatRecordSettings(payload: ChatRecordSettingsPayload) {
  const response = await axios.post<ApiResponse<{ ok: boolean; provider: ChatRecordProvider; status: string; message: string }>>(
    "/api/chat-records/settings/test",
    payload
  );
  return response.data.data;
}

export async function fetchChatRecordStatus(): Promise<ChatRecordStatus> {
  const response = await axios.get<ApiResponse<ChatRecordStatus>>("/api/chat-records/status");
  return response.data.data;
}

export async function fetchChatrooms(keyword?: string, talkers?: string[]): Promise<Chatroom[]> {
  const response = await axios.get<ApiResponse<Chatroom[]>>("/api/chat-records/chatrooms", {
    params: {
      keyword: keyword?.trim() || undefined,
      talkers: talkers && talkers.length ? talkers.join(",") : undefined
    }
  });
  return response.data.data;
}

File: src/app/schemas/chat_records.py
"""Schemas for chat record source settings and API responses."""

from typing import Literal, Optional

from pydantic import BaseModel, Field

from .base import ORMBase

ChatRecordProvider = Literal["chatlog", "weflow"]


class ChatRecordSettingsBase(BaseModel):
    provider: ChatRecordProvider = "chatlog"
    chatlog_base_url: str = "http://127.0.0.1:5030"
    chatlog_timeout_sec: int = Field(default=60, ge=1)
    chatlog_decrypt_before_fetch: bool = True
    chatlog_decrypt_cache_enabled: bool = True
    chatlog_decrypt_timeout_sec: int = Field(default=300, ge=1)
    chatlog_decrypt_cache_buffer_sec: int = Field(default=0, ge=0)
    chatlog_work_dir: str = ""
    weflow_base_url: str = "http://127.0.0.1:5031"
    weflow_token: Optional[str] = None
    weflow_page_limit: int = Field(default=1000, ge=1, le=10000)
    weflow_page_timeout_sec: int = Field(default=180, ge=1)
    weflow_empty_page_retry: int = Field(default=2, ge=0, le=10)


class ChatRecordSettingsUpdate(ChatRecordSettingsBase):
    pass


class ChatRecordSettingsOut(ORMBase):
    provider: ChatRecordProvider
    chatlog_base_url: str
    chatlog_timeout_sec: int
    chatlog_decrypt_before_fetch: bool
    chatlog_decrypt_cache_enabled: bool
    chatlog_decrypt_timeout_sec: int
    chatlog_decrypt_cache_buffer_sec: int
    chatlog_work_dir: str
    weflow_base_url: str
    has_weflow_token: bool = False
    weflow_page_limit: int
    weflow_page_timeout_sec: int
    weflow_empty_page_retry: int
    weflow_include_media: bool = False


class ChatRecordStatus(BaseModel):
    provider: ChatRecordProvider
    status: Literal["ok", "unreachable", "invalid_token", "unknown"]
    message: str


class ChatRecordTestResult(BaseModel):
    ok: bool
    provider: ChatRecordProvider
    status: str
    message: str
